alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -103,39 +107,75 @@ class TransportGetAlertsAction @Inject constructor(
             .sort(sortBuilder)
             .size(tableProp.size)
             .from(tableProp.startIndex)
+        var alertIndex = AlertIndices.ALL_ALERT_INDEX_PATTERN
+        if (!getAlertsRequest.alertIndex.isNullOrEmpty()) {
+            alertIndex = getAlertsRequest.alertIndex
+            getAlerts(searchSourceBuilder, alertIndex, actionListener, user)
+        } else if (getAlertsRequest.monitorId != null) {
+            transportGetMonitorAction.execute(
+                GetMonitorRequest(
+                    getAlertsRequest.monitorId,
+                    -3L,
+                    RestRequest.Method.GET,
+                    FetchSourceContext.FETCH_SOURCE
+                ),
+                object : ActionListener<GetMonitorResponse> {
+                    override fun onResponse(getMonitorResponse: GetMonitorResponse) {
+                        if (getMonitorResponse.monitor != null) {
+                            alertIndex = getMonitorResponse.monitor!!.dataSources.alertsIndex
+                            getAlerts(searchSourceBuilder, alertIndex, actionListener, user)
+                        }
+                    }
+
+                    override fun onFailure(t: Exception) {
+                        actionListener.onFailure(t)
+                    }
+                }
+            )
+        } else {
+            getAlerts(searchSourceBuilder, alertIndex, actionListener, user)
```

Comment:
``getAlerts`` is being called 3 times here in this block. We should first resolve the index name and then make the final call.